react-search-component/my-app/src/SearchableList.tsx

```diff
@@ -0,0 +1,39 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+
+type Props = {
+  list: string[];
+  value: string;
+  setValue: Dispatch<SetStateAction<string>>;
+};
+
+export function SearchableList({ list }: { list: string[] }) {
+  const [value, setValue] = useState('');
+
+  return (
+    <>
+      <SearchBar list={list} value={value} setValue={setValue} />
+      <MakeList list={list} value={value} setValue={setValue} />
+    </>
+  );
+}
+
+function SearchBar({ value, setValue }: Props) {
+  return (
+    <input
+      type="text"
+      value={value}
+      onChange={(e) => {
+        const targetValue = e.target.value;
+        setValue(targetValue);
+      }}
+    />
+  );
+}
+
+function MakeList({ value, list }: Props) {
+  const filterList = list.filter((i) =>
```

Comment:
This works fine, but I prefer to put the filtering logic up in the parent, which makes this component less knowledgeable. As discussed in Q&A, keeping the logic out of these subcomponents is ideal (when you can manage it).